### PR TITLE
Cover entity extraction helpers

### LINKED ISSUE #1261: Unit-test the template / entity-ID extraction layer in util.py
## Why This Matters
The regex stack in `util.py:86-125` (`ENTITY_ID_PATTERN`, `ENTITY_ID_TEMPLATE_PATTERNS`) plus `extract_entities_from_template_regex` (line 499), `split_comma_separated_entity_ids` (line 628), and `extract_template_strings_from_config` (line 651) are the engine behind every `unknown_*_references` repair in `ectoplasms/automation/repairs/`, `ectoplasms/script/repairs/`, `ectoplasms/lovelace/repairs/`, etc. They are pure(-ish) functions with rich edge cases (comma-separated lists, nested templates, `states.domain.entity` direct access, `expand(...)` filters, IGNORED_ENTITY_DOMAINS prefix matching). Pure functions = highest test ROI: no `hass` fixture overhead, parametrized tables cover dozens of cases in milliseconds, and any future regex tweak (a recurring source of false-positive repairs — see commit `29fd59d`) gets immediate feedback.

## Approach
Add `tests/test_util.py` with three parametrized test classes: (a) `test_extract_entities_from_template_regex` over a table of ~20 templates including `{{ states('light.kitchen') }}`, `{{ state_attr('sensor.x', 'foo') }}`, `{{ expand('group.lights') }}`, `states.binary_sensor.door.state`, multi-entity strings, malformed templates, and false-positive bait like service names. (b) `test_split_comma_separated_entity_ids` covering empty/None/single/multi/whitespace edge cases. (c) `test_extract_template_strings_from_config` over dicts, lists, nested structures, and non-string values. For functions that take `hass` (e.g. `async_filter_known_entity_ids`), use a minimal `hass` fixture and seed `er.async_get(hass)` with known entities.

## Acceptance Criteria
- [ ] >=20 parametrized cases for `extract_entities_from_template_regex`, including known-false-positive bait
- [ ] `split_comma_separated_entity_ids` covers all branches in the function
- [ ] `IGNORED_ENTITY_DOMAINS` filtering is asserted (no `scene.*` or `group.*` leaks)
- [ ] At least one test reproduces the bug fixed by commit `29fd59d` ("Find unknown entities in heading badge") to prevent regression

## Risks & Caveats
These regexes are deliberately permissive to catch unknown entities — be careful not to write tests that lock down behavior the maintainer actually wants flexible. Pair with the maintainer on the table of cases. The `KNOWN_DOMAINS` list (`util.py:84`) is a hard-coded enum; tests should treat it as a fixture so domain additions don't silently break.

## Scores
- Impact: ██████░░░░ 6/10
- Difficulty: ███░░░░░░░ 3/10
- Short-Term ROI: █████████░ 9/10
- Long-Term Value: ███████░░░ 7/10

## Priority
Prototype First

## Dependencies
#1257

### PR DESCRIPTION
**Pull request by frenckatron**
## Description

Adds regression coverage for the entity extraction helpers used by unknown entity reference repairs.

This covers template entity extraction forms, service false-positive filtering, ignored entity domains, and the Lovelace heading card badge extraction path added by #1173.

## Motivation and Context

Closes #1261.

These helpers are regex-heavy and shared by multiple repair flows. A small parametrized test table gives quick feedback when future tweaks change what entities are discovered.

## How has this been tested?

- `uv run pytest tests/test_util.py tests/ectoplasms/lovelace/repairs/test_unknown_entity_references.py -q`
- `uv run ruff check tests/test_util.py tests/ectoplasms/lovelace/repairs/test_unknown_entity_references.py`
- `uv run ruff format --check tests/test_util.py tests/ectoplasms/lovelace/repairs/test_unknown_entity_references.py`
- `uv run pytest -q`

## Screenshots (if appropriate):

Not applicable.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: tests/test_util.py [hunk 1]
```diff
@@ -139,3 +144,82 @@ def test_extract_templates_appends_to_caller_supplied_list() -> None:
...
+        ),
+        (
+            "{{ ['light.kitchen'] | select('is_state', 'on') | list }}",
+            {"light.kitchen"},
+        ),
+        ("{{ 'light.' ~ room }}", set()),
+        ("{{ states('unknown_domain.foo') }}", set()),
+        ("{{ states('light.') }}", set()),
+        ("{{ 'homeassistant.turn_on' }}", set()),
+    ],
```
[the remark this answers is not in the document]
Comment (frenckatron) [reply tests/test_util.py:192]: Fixed. The false-positive case now uses `light.turn_on`, which matches the known entity domain pattern and is then removed through the known service subtraction.

File: tests/test_util.py [hunk 2]
```diff
@@ -139,3 +144,82 @@ def test_extract_templates_appends_to_caller_supplied_list() -> None:
...
+    hass: HomeAssistant,
+    template: str,
+    expected: set[str],
+) -> None:
+    """Test entity IDs are extracted from supported template forms."""
+    hass.services.async_register(
+        "homeassistant",
+        "turn_on",
+        lambda _: None,
+    )
```
[the remark this answers is not in the document]
Comment (frenckatron) [reply tests/test_util.py:204]: Fixed. The test now registers `light.turn_on` explicitly, so the service-filtering assertion is deterministic.

File: tests/test_util.py [hunk 3]
```diff
@@ -139,3 +144,82 @@ def test_extract_templates_appends_to_caller_supplied_list() -> None:
...
+) -> None:
+    """Test ignored entity domains do not leak as unknown template entities."""
+    unknown = await async_filter_known_entity_ids_with_templates(
+        hass,
+        {
+            "{{ states('scene.goodnight') }}",
+            "{{ states('group.family') }}",
+            "{{ states('device_tracker.phone') }}",
+            "{{ states('persistent_notification.update') }}",
+            "{{ states('light.missing') }}",
```
[the remark this answers is not in the document]
Comment (frenckatron) [reply tests/test_util.py:220]: Fixed. `persistent_notification.update` is now passed as a raw entity ID so it exercises the non-template ignored-domain filtering path.

**Review by frenck — APPROVED** [review]
